src/FSharpPlus/Extensions/Task.fs

```diff
@@ -21,6 +21,18 @@ module Task =
         x.ContinueWith(fun (x' : Task<'T>) ->
             y.ContinueWith(fun (y' : Task<'U>) ->
                 (f x'.Result y'.Result))).Unwrap()
+        
+    /// <summary>Creates a task workflow from three workflows 'x', 'y' and z, mapping its results with 'f'.</summary>
+    /// <remarks>Workflows are run in sequence.</remarks>
+    /// <param name="f">The mapping function.</param>
+    /// <param name="x">First task workflow.</param>
+    /// <param name="y">Second task workflow.</param>
+    /// <param name="z">Third task workflow.</param>
+    let map3 (f : 'T -> 'U -> 'V -> 'W) (x : Task<'T>) (y : Task<'U>) (z: Task<'V>) : Task<'W> =
+        x.ContinueWith(fun (x' : Task<'T>) ->
+            y.ContinueWith(fun (y' : Task<'U>) ->
+                z.ContinueWith(fun (z' : Task<'V>) ->
+                (f x'.Result y'.Result z'.Result)))).Unwrap().Unwrap()
```

Comment:
Note that this implementation is not very useful as it doesn't handle errors for instance.
I know at the time you started this PR this was the state of things, but if you look at master now, you can see how `zip` and `map2` look like and get an idea of how this should be properly implemented and tested.